stdlib/auth/src/main/java/org/ballerinalang/auth/ldap/util/LDAPUtils.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.auth.ldap.util;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.ballerinalang.auth.ldap.CommonLDAPConfiguration;
+import org.ballerinalang.auth.ldap.LDAPConnectionContext;
+import org.ballerinalang.auth.ldap.UserStoreException;
+
+import javax.naming.CompositeName;
+import javax.naming.InvalidNameException;
+import javax.naming.Name;
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.SearchControls;
+import javax.naming.directory.SearchResult;
+import javax.naming.ldap.LdapName;
+
+/**
+ * Utility class for LDAP related common operations.
+ */
+public class LDAPUtils {
+
+    private static final Log LOG = LogFactory.getLog(LDAPUtils.class);
+
+    /**
+     * Checks whether a given string is null or empty after the trim.
+     *
+     * @param str String which needs to be check for null or empty
+     * @return true if the string is null or empty else false
+     */
+    public static boolean isNullOrEmptyAfterTrim(String str) {
+        return str == null || str.trim().isEmpty();
+    }
+
+    /**
+     * This is to search user and retrieve ldap name directly from ldap.
+     *
+     * @param userName          Given username
+     * @param ldapConfiguration LDAP user store configurations
+     * @param connectionSource  LDAPConnectionContext
+     * @return Associated name for the given username
+     * @throws UserStoreException if there is any exception occurs during the process
+     */
+    public static String getNameInSpaceForUsernameFromLDAP(String userName, CommonLDAPConfiguration ldapConfiguration,
+                                  LDAPConnectionContext connectionSource) throws UserStoreException {
+
+        String searchBase = null;
```

Comment:
Seems like we don't need to declare the variable here. We can declare and initialize it in line 68.